src/main/starlark/builtins_bzl/common/cc/cc_shared_library_wrapper.bzl

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+load(":common/cc/cc_shared_library.bzl", _cc_shared_library = "cc_shared_library")
+
+"""Implementation of cc_shared_library's macro wrapper"""
+
+def cc_shared_library(**kwargs):
```

Comment:
This being a macro would require migrating the attribute `doc`s to a function comment.

@comius What do you think of allowing `native.package_relative_label` in `initializer`s instead, perhaps by providing a suitably trimmed down `BazelStarlarkContext` to the thread running the `initializer`? I understand why you don't want `native.existing_rules` to be run in one, but label conversion will be a frequent ask with Bzlmod.

Reply:
Initializers already lift strings of regular labels. Maybe the solution is to make `exports_filter` a label?

Should the lifting really be package relative? That's only the property of output labels. Otherwise labels are lifted call-site relative. Not much difference at the moment. But if you had a string label in an initializer, that would need to work relative to the initializer, right?

Reply:
I don’t want to expose such functions to initializers, because they open space for crafting weird conversions and bugs with that.

A new type like attr.nodep_label_list would be the right thing to do. We have it natively.

But I’m also wondering, what’s going on in cc_shared_library, to make a usecase like this. In all the rules we touched, label comparison was a bad idea and we removed it.

Reply:
Yeah, I think it's better to expose `package_relative_label` to initializers and use them.

Without initializers also the docs on bazel.build would break.

Reply:
I implemented this in https://github.com/bazelbuild/bazel/pull/21740. I also allowed the other non-target related functions as they seemed equally safe to allow in initializers.

Reply:
I am now using an initializer, PTAL.